### PR TITLE
[Ide] TreeView now takes into account different font sizes

### PR DESCRIPTION
**Pull request by Therzok**
Bug 37814 - Unnecessary tooltip in solution pad

### CODE REVIEW

**Comment by Therzok** [comment]
@slluis The previous behaviour would calculate a row's width based on the font size at the construction of the widget - which is later changed by setting it as per Ide Theme. With this change, the layout now takes into account 'zooming' which changes font size and fixes the width calculation on Mac, where the font size is 11 (in the layout - as opposed to the Widget's context with 12).

**Comment by slluis** [comment]
Hmm, changing the tree font every time a cell is rendered doesn't look right to me. The problem seems to be that the cell size is not properly calculated, in which case it should be fixed in the GetSize() override.

**Comment by Therzok** [comment]
Does this look better?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/ExtensibleTreeView.cs [hunk 1]
```diff
@@ -2494,6 +2499,19 @@ public Gdk.Rectangle GetStatusIconArea (Gtk.Widget widget, Gdk.Rectangle cell_ar
 			public override void GetSize (Gtk.Widget widget, ref Gdk.Rectangle cell_area, out int x_offset, out int y_offset, out int width, out int height)
 			{
 				base.GetSize (widget, ref cell_area, out x_offset, out y_offset, out width, out height);
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/ExtensibleTreeView.cs]: We have all the information to calculate the size, so calling base.GetSize() shouldn't be necessary. All we need to do is use the layout to calculate the size of the text and add the margins.

**Comment by Therzok** [comment]
Superseded by https://github.com/mono/monodevelop/pull/1354

**Pull request closed without merging**